src/geode/inspector/topology/private/brep_blocks_topology_impl.cpp

```diff
@@ -51,9 +54,109 @@ namespace geode
         bool BRepBlocksTopologyImpl::brep_vertex_blocks_topology_is_valid(
             index_t unique_vertex_index ) const
         {
-            const auto block_uuids =
-                components_uuids( brep_.component_mesh_vertices(
-                    unique_vertex_index, Block3D::component_type_static() ) );
+            const auto block_cmvs = brep_.component_mesh_vertices(
+                unique_vertex_index, Block3D::component_type_static() );
+            const auto block_uuids = components_uuids( block_cmvs );
+            if( block_uuids.size() == 1 )
+            {
+                const auto line_cmvs = brep_.component_mesh_vertices(
+                    unique_vertex_index, Line3D::component_type_static() );
+                const auto surface_cmvs = brep_.component_mesh_vertices(
+                    unique_vertex_index, Surface3D::component_type_static() );
```

Comment:
They are also used if size is 2, you can initialize them before the if statement